WooCommerce/Classes/ViewRelated/Dashboard/DashboardViewController.swift

```diff
@@ -178,6 +211,36 @@ private extension DashboardViewController {
             self?.onDashboardUIUpdate(forced: forced, updatedDashboardUI: dashboardUI)
         })
     }
+
+    /// Display the error banner at the top of the dashboard content (below the site title)
+    ///
+    func showTopBannerView() {
+        guard let dashboardUI = dashboardUI else {
+            return
+        }
+        stackView.addArrangedSubview(topBannerView)
+        stackView.addArrangedSubview(spacerView)
+        if !shouldShowStoreNameAsSubtitle {
+            containerView.addSubview(stackView)
+            dashboardUI.view.translatesAutoresizingMaskIntoConstraints = false
+            NSLayoutConstraint.activate([
+                stackView.topAnchor.constraint(equalTo: containerView.topAnchor),
+                stackView.leadingAnchor.constraint(equalTo: containerView.leadingAnchor),
+                stackView.trailingAnchor.constraint(equalTo: containerView.trailingAnchor),
+                dashboardUI.view.topAnchor.constraint(equalTo: stackView.bottomAnchor),
+                dashboardUI.view.leadingAnchor.constraint(equalTo: containerView.leadingAnchor),
+                dashboardUI.view.trailingAnchor.constraint(equalTo: containerView.trailingAnchor),
+                dashboardUI.view.bottomAnchor.constraint(equalTo: containerView.bottomAnchor)
+            ])
+        }
+    }
+
+    /// Hide the error banner
+    ///
+    func hideTopBannerView() {
+        topBannerView.removeFromSuperview()
+        spacerView.removeFromSuperview()
```

Comment:
I think there's no need for adding and removing `topBannerView` and `spacerView` when showing / hiding the banner view. The great thing about stack view is that it can collapse or expand when showing / hiding its subviews. 

Therefore, I'd suggest that you add these as arranged subviews  right after adding the `innerStackView`, and hide them by default. When showing or hiding the banner view, all you need to do is show / hide these subviews and it should still work as expected.

Reply:
This is a great approach that I didn't think of, thanks! In 92e6c6c I changed how these views are configured, and it simplifies the way the constraints are set, too. Now the header stack view is configured up front with all the arranged subviews added, and as you suggested the methods to show or hide the banner just show/hide those subviews.